### PR TITLE
adding contributing.md to snowflake repo

### PR DESCRIPTION
**Pull request by McKnight-42**
### Description
Adding Contributing.md file to each seperated adapter

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-snowflake next" section.

### CODE REVIEW

**Comment by VersusFacit** [comment]
Really love to this @McKnight-42. Any reason we stalled out here? Anything I can do to help? 😄

**Comment by McKnight-42** [comment]
@VersusFacit i'm doing the bigquery one first to get a template for all the other adapters so moved made one for each at the base portion I had for bigquery and will update them based on all changes made to that one as they should generally be the same just updating tests examples and names.

**Comment by VersusFacit** [comment]
Great. Glad we have this context logged. Good to know this is still very much in the works 🖖

**Comment by McKnight-42** [comment]
I agree and on that point i will add this here [bigquery contributing.md](https://github.com/dbt-labs/dbt-bigquery/pull/73)

File: CONTRIBUTING.MD [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+When `dbt-snowflake` is installed this way, any changes you make to the `dbt-snowflake` source code will be reflected immediately in your next `dbt-snowflake` run.
+
+To confirm you have correct version of `dbt-core` installed please run `dbt --version` and `which dbt`.
+
+
+## Testing
+
+### Initial Setup
+
+`dbt-snowflake` contains [unit](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/unit) and [integration](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/integration) tests. Integration tests require tsting against an actual Snowflake warehouse. We have CI set up to test against a Snowflake warehouse. In order to run integration tests locally, you will need a `test.env` file in the root of the repository that contains credentials for Snowflake.
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:58]: got a little typo here

And for some reason, I struggle to read this. How about we just modify the text so that we introduce the idea you can run tests locally on on our CI system for pull requests. Then, we instruct those who want to locally test how to do so in maybe even a separate line. It's a bit jampacked right now. *takes off editor cap*

File: CONTRIBUTING.MD [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+1. [About this document](#about-this-document)
+3. [Getting the code](#getting-the-code)
+5. [Running `dbt-snowflake` in development](#running-dbt-snowflake-in-development)
+6. [Testing](#testing)
+7. [Updating Docs](#updating-docs)
+7. [Submitting a Pull Request](#submitting-a-pull-request)
+
+## About this document
+This document is a guide intended for folks interested in contributing to `dbt-snowflake`. Below, we document the process by which members of the community should create issues and submit pull requests (PRs) in this repository. It is not intended as a guide for using `dbt-snowflake`, and it assumes a certain level of familiarity with Python concepts such as virtualenvs, `pip`, python modules, filesystems, and so on. This guide assumes you are using macOS or Linux and are comfortable with the command line.
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:11]: Could we actually point people to the dbt docs for using dbt-snowflake maybe :) 

Also, one thing I fail at but try to work at is writing shorter sentences. Let's break up the long one in here!

File: CONTRIBUTING.MD [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+For those wishing to contribute we highly suggest reading the [dbt-core](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md). if you haven't already. Almost all of the information there is applicable to contributing here, too!
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:13]: text formatting issues here

While you're mucking around in this, how about just removing the 'if you haven't already' clause. Let's keep it short and snappy where there isn't essential information.

File: CONTRIBUTING.MD [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+- Docs are [here](https://docs.getdbt.com/).
+- The docs repo for making changes is located [here]( https://github.com/dbt-labs/docs.getdbt.com).
+- The changes made are likely to impact one or both of [Snowflake Profile](https://docs.getdbt.com/reference/warehouse-profiles/snowflake-profile), or [Snowflake Configs](https://docs.getdbt.com/reference/resource-configs/snowflake-configs).
+- We ask every community member who makes a user-facing change to open an issue or PR regarding doc changes.
+
+
+
+## Submitting a Pull Request
+
+dbt Labs provides a CI environment to test changes to the `dbt-snowflake` adapter, and periodic checks against the development version of `dbt-core` through Github Actions.
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:97]: what does this periodic checks part mean?

File: CONTRIBUTING.MD [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+- We ask every community member who makes a user-facing change to open an issue or PR regarding doc changes.
+
+
+
+## Submitting a Pull Request
+
+dbt Labs provides a CI environment to test changes to the `dbt-snowflake` adapter, and periodic checks against the development version of `dbt-core` through Github Actions.  
+
+A `dbt-snowflake` maintainer will review your PR. They may suggest code revision for style or clarity, or request that you add unit or integration test(s). These are good things! We believe that, with a little bit of help, anyone can contribute high-quality code.
+
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:100]: I feel like there should be a comment in this that mentions needing a dbt employee to approve the CI workflow somewhere around here.

**Review by VersusFacit — CHANGES_REQUESTED** [review]
Nice guide so far. Leaving some comments.

File: CONTRIBUTING.MD (hunk 4 above)
Comment (McKnight-42) [reply CONTRIBUTING.MD:101]: I believe this is our nightly checks? not exactly sure if we have any triggers between the two otherwise maybe releases? times we would need to make sure changes to one doesn't break the other.

File: CONTRIBUTING.MD [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+### Installation
+
+First make sure that you set up your `virtualenv` as described in [Setting up an environment](https://github.com/dbt-labs/dbt-core/blob/HEAD/CONTRIBUTING.md#setting-up-an-environment).  Ensure you have the latest version of `pip` installed with `pip install --upgrade pip`. Next, install `dbt-snowflake` latest dependencies:
+
+```sh
+pip install -e . -r dev-requirements.txt
+```
+
+When `dbt-snowflake` is installed this way, any changes you make to the `dbt-snowflake` source code will be reflected immediately in your next `dbt-snowflake` run.
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:51]:
```suggestion
By installing `dbt-snowflake` this way, `dbt` uses the source code in your local `dbt-snowflake` repository exactly as it is when you invoke the program. Saved changes to source code are reflected in the next `dbt` run.
```
A suggestion to make this more explicit (to me! Maybe you can do better than what I've come up with.)

File: CONTRIBUTING.MD [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+
+To confirm you have correct version of `dbt-core` installed please run `dbt --version` and `which dbt`.
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:53]:
```suggestion
```

File: CONTRIBUTING.MD [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+
+## Testing
+
+### Initial Setup
+
+`dbt-snowflake` contains [unit](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/unit) and [integration](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/integration) tests. Integration tests require testing against an actual Snowflake warehouse. We have CI set up to test against a Snowflake warehouse during PR checks.
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:60]:
```suggestion
`dbt-snowflake` contains [unit](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/unit) and [integration](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/integration) tests. Integration tests require testing against an actual Snowflake warehouse. There are two main options for running tests:

- This repository has a CI pipeline active. It runs tests against a Snowflake warehouse during PR checks. 
- You may also run integration tests locally. This requires a `test.env` file in the root of your snowflake repository. This file must contain credentials for accessing a valid Snowflake account.
```
Figured it might be beneficial to more clearly group and format testing methods.

File: CONTRIBUTING.MD [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+To confirm you have correct version of `dbt-core` installed please run `dbt --version` and `which dbt`.
+
+
+## Testing
+
+### Initial Setup
+
+`dbt-snowflake` contains [unit](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/unit) and [integration](https://github.com/dbt-labs/dbt-snowflake/tree/main/tests/integration) tests. Integration tests require testing against an actual Snowflake warehouse. We have CI set up to test against a Snowflake warehouse during PR checks. 
+
+In order to run integration tests locally, you will need a `test.env` file in the root of the repository that contains credentials for Snowflake.
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:62]:
```suggestion
```

File: CONTRIBUTING.MD [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+
+Note: This `test.env` file is git-ignored, but please be _extra_ careful to never check in credentials or other sensitive information when developing. To create your `test.env` file, copy the provided example file, then supply your relevant credentials.
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:64]:
```suggestion
Note: This `test.env` file is git-ignored, but please be _extra_ careful to never check in credentials or other sensitive information when developing. To create your `test.env` file, copy the example file provided in this repository, then fill-in your relevant credentials.
```

File: CONTRIBUTING.MD [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+```
+cp test.env.example test.env
+$EDITOR test.env
+```
+
+### Test commands
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:71]:
```suggestion
### More on running test commands locally
```

File: CONTRIBUTING.MD [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+There are a few methods for running tests locally.
+
+#### `tox`
+`tox` takes care of managing Python virtualenvs and installing dependencies in order to run tests. You can also run tests in parallel, for example you can run unit tests for Python 3.7, Python 3.8, Python 3.9, and `flake8` checks in parallel with `tox -p`. Also, you can run unit tests for specific python versions with `tox -e py37`. The configuration of these tests are located in `tox.ini`.
+
+#### `pytest`
+Finally, you can also run a specific test or group of tests using `pytest` directly. With a Python virtualenv active and dev dependencies installed you can do things like:
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:78]:
```suggestion
You can also run a specific test or group of tests using `pytest` directly. With a Python virtualenv active and dev dependencies installed, you can run tests with commands like:
```

File: CONTRIBUTING.MD [hunk 13]
```diff
@@ -0,0 +1,105 @@
...
+# run specific snowflake integration tests
+python -m pytest -m profile_snowflake tests/integration/simple_copy_test
+# run all unit tests in a file
+python -m pytest tests/unit/test_snowflake_adapter.py
+# run a specific unit test
+python -m pytest tests/unit/test_snowflake_adapter.py::TestSnowflakeAdapter::test_convert_date_type
+```
+## Updating Docs
+
+Many changes will require an update to the `dbt-snowflake` docs here are some useful resources.
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:90]:
```suggestion
Many changes will require an update to `dbt-snowflake` documentation. Here are some relevant links.
```

Grammar

File: CONTRIBUTING.MD (hunk 4 above)
Comment (VersusFacit) [reply CONTRIBUTING.MD:101]: I'm pretty sure if anything this would refer to checks whenever a commit is pushed as opposed to referring to our nightly monitoring checks. And hence, I'd suggest rewording this to be something like, "As mentioned above, dbt Labs....adapter. The GitHub Actions pipeline runs your feature branch using the current development version of `dbt-core`" -- something that indicates the CI pipeline uses the main branch in dbt core.

**Review by VersusFacit — CHANGES_REQUESTED** [review]
Left a bunch of comments now that I feel prepared enough to offer meaningful comments here! A lot of these where I gave a reasoned comment are of the p0 variety, where I think they should be addressed in some fashion, whether by the suggestions I wrote or with something inspired by them but better in your estimate!  

Smaller comments are nitpicks for clarity, so take them or ignore them as your judgement sees fit. Do feel empowered to push back on any them. I'll keep an eye on this PR so we can get it merged.

File: CONTRIBUTING.MD [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+#### `pytest`
+You can also run a specific test or group of tests using `pytest` directly. With a Python virtualenv active and dev dependencies installed you can run test commands like:
+
+```sh
+# run specific snowflake integration tests
+python -m pytest -m profile_snowflake tests/integration/simple_copy_test
+# run all unit tests in a file
+python -m pytest tests/unit/test_snowflake_adapter.py
+# run a specific unit test
+python -m pytest tests/unit/test_snowflake_adapter.py::TestSnowflakeAdapter::test_convert_date_type
```
Comment (VersusFacit) [inline CONTRIBUTING.MD:92]: also add an integration test example or just say this can either be unit or integration

**Review by VersusFacit — APPROVED** [review]
This one got bogged down in revisions and open asks. Matt did great work here (thanks Matt!). 

To make sure this doesn't drag any longer, I put in a little time to get it pushed over the finish line. We can iterate on it as we see fit from here on. Also, I have let the community know about this
